src/binary-reader-ir.cc

```diff
@@ -795,6 +795,7 @@ Result BinaryReaderIR::OnAtomicLoadExpr(Opcode opcode,
                                         Index memidx,
                                         Address alignment_log2,
                                         Address offset) {
+  module_->features_used.atomic_ops = true;
```

Comment:
I think it _might_ be possible to detect this in one place in BinaryReaderIR::OnOpcode (by checking if opcode.GetPrefix() = 0xfe, or checking if the opcode is in the range from MemoryAtomicNotify to I64AtomicRmw32CmpxchgU) rather than in the individual expr callbacks.

Reply:
Or if you wanted to skip the feature _tracking_ and just restore the Features member of the wasm2c options and make it included based on whether `--enable-threads` is given (under the assumption that this won't be the default for a long time), also fine by me.